src/Symfony/Component/Yaml/Inline.php

```diff
@@ -627,7 +627,7 @@ public static function evaluateBinaryScalar($scalar)
 
     private static function isBinaryString($value)
     {
-        return preg_match('/[^\x09-\x0d\x20-\xff]/', $value);
+        return !preg_match('//u', $value);
```

Comment:
Oh sorry, I missed your second comment. The previous regex was borrowed from PHPUnit. But if you have good experience with the other one, I am fine with using it.